pkg/client/record/event.go

```diff
@@ -86,6 +89,9 @@ type EventBroadcaster interface {
 	// NewRecorder returns an EventRecorder that can be used to send events to this EventBroadcaster
 	// with the event source set to the given event source.
 	NewRecorder(source api.EventSource) EventRecorder
+
+	// NewRecorderWithClock makes it easy to write test with a fake clock.
+	NewRecorderWithClock(source api.EventSource, clock util.Clock) EventRecorder
```

Comment:
Oh wow this is gross. I see why you have to do it, and that's bad. This would be so much better if `NewBroadcaster` return an implementation and not an interface. If it did, you could add this as a private method to that implementation, access it in the tests, and never expose it to anything outside this package. Would it be hard to change `eventBroadcasterImpl` to a public type (probably a different name), return that from `NewBroadcaster`, and then just have this be a private method?

This is a good example of why you shouldn't return interfaces if you can.